cmd/gitops/check/oidcconfig/cmd.go

```diff
@@ -0,0 +1,127 @@
+			"on the cluster.",
+		Example: `
+# Check the OIDC configuration stored in the flux-system/oidc-auth Secret
+gitops check oidc-config
+
+# Check a different set of scopes
+gitops check oidc-config --scopes=openid,groups
+
+# Check a different username cliam
+gitops check oidc-config --claim-username=sub
+
+# Check configuration without fetching a Secret from the cluster
+gitops check oidc-config --skip-secret --client-id=CID --client-secret=SEC --issuer-url=https://example.org
+		`,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if skipSecretFlag {
+				fromSecretFlag = "" // the skip flag overrides this one.
+			}
+
+			cfg, err := kubeConfigArgs.ToRESTConfig()
+			if err != nil {
+				return err
+			}
+
+			if fromSecretFlag == "" {
+				if clientIDFlag == "" ||
+					clientSecretFlag == "" ||
+					issuerURLFlag == "" {
+					return fmt.Errorf("when not reading OIDC configuration from a Secret, you need to provide " +
+						"client ID, client secret and issuer URL using the respective command-line flags")
+				}
+			}
+
+			log := logger.NewCLILogger(os.Stdout)
+			kubeClient, err := run.GetKubeClient(log, *kubeConfigArgs.Context, cfg, nil)
+			if err != nil {
+				return cmderrors.ErrGetKubeClient
+			}
+
+			ns, err := cmd.Flags().GetString("namespace")
+			if err != nil {
+				return fmt.Errorf("failed getting namespace flag: %w", err)
+			}
+
+			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+			defer cancel()
+			claims, err := check.GetClaims(ctx, check.Options{
+				ClientID:        clientIDFlag,
+				ClientSecret:    clientSecretFlag,
+				IssuerURL:       issuerURLFlag,
+				SecretName:      fromSecretFlag,
+				SecretNamespace: ns,
+				Scopes:          scopesFlag,
+				ClaimUsername:   claimUsernameFlag,
+			}, log, kubeClient)
+
+			if err != nil {
+				return fmt.Errorf("failed getting claims: %w", err)
+			}
+
+			log.Println("user: %s", claims.Username)
+			if len(claims.Groups) != 0 {
+				log.Println("groups: %s", strings.Join(claims.Groups, ", "))
+			} else {
+				log.Println("no groups claim")
+			}
+
+			return nil
+		},
+		DisableAutoGenTag: true,
+	}
+
+	kubeConfigArgs = run.GetKubeConfigArgs()
+	kubeConfigArgs.AddFlags(cmd.Flags())
+
+	cmd.Flags().StringVar(&clientIDFlag, "client-id", "", "OIDC client ID")
+	cmd.Flags().StringVar(&clientSecretFlag, "client-secret", "", "OIDC client secret")
+	cmd.Flags().StringVar(&issuerURLFlag, "issuer-url", "", "OIDC issuer URL")
+	cmd.Flags().StringVar(&fromSecretFlag, "from-secret", "oidc-auth", "Get OIDC configuration from the given Secret resource")
+	cmd.Flags().BoolVar(&skipSecretFlag, "skip-secret", false,
+		"Do not read OIDC configuration from a Kubernetes Secret but rely solely on the values from the given flags.")
+	cmd.Flags().StringVar(&claimUsernameFlag, "claim-username", "", "ID token claim to use as the user name.")
```

Comment:
How about using `username-claim` which would mirror the K8s options?

(the other fields seem to match up).

>  `--oidc-username-claim string     Default: "sub"`

(_if_ you do this, make sure you fix the examples )

We (and K8s) allow you to configure the groups claim too.

Reply:
Renamed the existing and added a `--groups-claim` flag, too.